bundles/org.openhab.binding.hpprinter/src/main/java/org/openhab/binding/hpprinter/internal/HPPrinterHandler.java

```diff
@@ -116,13 +117,16 @@ public void channelUnlinked(ChannelUID channelUID) {
         }
     }
 
+    @Override
     protected void updateStatus(final ThingStatus status, final ThingStatusDetail thingStatusDetail,
             @Nullable final String message) {
         super.updateStatus(status, thingStatusDetail, message);
     }
 
     public void updateState(final String group, final String channel, final State state) {
-        super.updateState(new ChannelUID(thing.getUID(), group, channel), state);
+        if (thing.getStatus() == ThingStatus.ONLINE) {
```

Comment:
This does not look like a proper fix to me.
In general, it should be possible to send updates, even if the thing status is UNKNOWN or OFFLINE.
It would imho be better to fix the root cause: That some logic keeps running and isn't interrupted when `dispose()` returns.

Reply:
@kaikreuzer I am rather obsessive about clean log files so I wanted to make a one line fix :) but you are right that it is not a "proper" solution :( If you want that, then it will take me a bit longer. Please wait..

Reply:
> I am rather obsessive about clean log files 

That's a very valid obsession, which I like and fully understand 😎 

> If you want that, then it will take me a bit longer.

Additional time granted. ;-) Take as long as you need, I prefer proper solutions over quick hacks.

Reply:
I just pushed some new commits that a) roll back my prior hack, and b) attempt to solve it a different way.

>some logic keeps running and isn't interrupted

Actually the logic **was** interrupting the schedulers (via scheduler.cancel()) however there were already started jetty client HTTP calls in the pipeline that had not yet completed. My fix does not actually interrupt those jetty HTTP calls, because I didn't want to risk messing up something on the printer. Instead it lets such calls complete (or timeout), and just shortcuts any further processing of the (eventual) HTTP responses.